packages/ref-core/src/cmip_ref_core/providers.py

```diff
@@ -86,3 +116,233 @@ def get(self, slug: str) -> Metric:
+    def prefix(self, path: Path) -> None:
+        self._prefix = path
+
+    def configure(self, config: Config) -> None:
+        """Configure the provider."""
+        self.prefix = config.paths.software / "conda"
+
+    def _install_conda(self, update: bool) -> Path:
+        """Install micromamba in a temporary location."""
+        self.prefix.mkdir(parents=True, exist_ok=True)
+        conda_exe = self.prefix / "micromamba"
+        if not conda_exe.exists():
+            logger.info("Installing conda")
+            response = requests.get(_get_micromamba_url(), timeout=120)
+            response.raise_for_status()
+            with conda_exe.open(mode="wb") as file:
+                file.write(response.content)
+            conda_exe.chmod(stat.S_IRWXU)
+            logger.info("Successfully installed conda.")
+        elif update:
+            logger.info("Updating conda")
+            subprocess.run([str(conda_exe), "self-update"], check=True)  # noqa: S603
+            logger.info("Successfully updated conda")
+
+        return conda_exe
+
+    def get_conda_exe(self, update: bool = False) -> Path:
+        """
+        Get the path to a conda executable.
+        """
+        if self._conda_exe is None:
+            self._conda_exe = self._install_conda(update)
+        return self._conda_exe
+
+    def get_environment_file(self) -> AbstractContextManager[Path]:
+        """
+        Return a context manager that provides the environment file as a Path.
+        """
+        # Because providers are instances, we have no way of retrieving the
+        # module in which they are created, so get the information from the
+        # first registered metric instead.
+        metrics = self.metrics()
+        if len(metrics) == 0:
+            msg = "Unable to determine the provider module, please register a metric first."
+            raise ValueError(msg)
+        module = metrics[0].__module__.split(".")[0]
+        lockfile = importlib.resources.files(module).joinpath("requirements").joinpath("conda-lock.yml")
+        return importlib.resources.as_file(lockfile)
+
+    @property
+    def env_path(self) -> Path:
+        """
+        A unique path for storing the conda environment.
+        """
+        with self.get_environment_file() as file:
+            suffix = hashlib.sha1(file.read_bytes(), usedforsecurity=False).hexdigest()
+        return self.prefix / f"{self.name}-{self.version}-{suffix}"
+
+    def create_env(self) -> None:
+        """
+        Create a conda environment.
+        """
+        if self.env_path.exists():
+            logger.info(f"Environment at {self.env_path} already exists, skipping.")
+            return
+
+        with self.get_environment_file() as file:
+            cmd = [
+                f"{self.get_conda_exe(update=True)}",
+                "create",
+                "--yes",
+                "--file",
+                f"{file}",
+                "--prefix",
+                f"{self.env_path}",
```

Comment:
I had to add "--platform osx-64" to support the pmp environment. Perhaps we need to add an optional platform configuration value in the Provider configuration to enable overriding the default platform?

Reply:
Maybe pmp or one of its dependencies is not available on conda-forge for your default platform, see https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html for some background. Shall we try to tackle this in a follow up pull request? An alternative solution could be to make sure that all packages are available for all the platforms we support.

Reply:
I looked into this a bit further: trying to solve the pcmdi_metrics environment for `arm64` results in `"nothing provides cdms2 needed by pcmdi_metrics-3.0.1-pyhd8ed1ab_0"`. The cdms package [is no longer maintained](https://github.com/CDAT/cdms) and it looks like @lee1043 is working on getting rid of the dependency https://github.com/PCMDI/pcmdi_metrics/issues/1203, so this problem will hopefully be solved soon.

Reply:
Looking through the [`mamba` source code](https://github.com/mamba-org/mamba/blob/26785a6b50bea434cf8b7d964f92f0f407b34b0a/libmamba/src/api/configuration.cpp#L1372-L1380), it looks like the platform can be specified by using the environmental variables `CONDA_SUBDIR` or `MAMBA_PLATFORM`. Maybe that is a sufficiently good solution for now, since in the longer run this problem will be solved by removing the cdms dependency.

Reply:
That will work. For Arm-based mac users, they will need to manually install the pmp environment via

`MAMBA_PLATFORM=osx-arm64 ref providers create-env --provider pmp`

I'll add that to the docs for the follow up PR for pmp